apollo-router/src/apollo_studio_interop/mod.rs

```diff
@@ -486,28 +502,33 @@ impl UsageGenerator<'_> {
         is_referenced: bool,
         is_null_reference: bool,
     ) {
-        match self
-            .input_field_references
-            .entry(type_name)
-            .or_default()
-            .entry(field_name)
-        {
-            Entry::Occupied(mut entry) => {
-                // The input object field stats are additive in that it represents whether the field was referenced/null/undefined
-                // at least once within this operation.
-                let stats: &mut InputObjectFieldStats = entry.get_mut();
+        // Not using entry API here due to performance impact
+        let type_name_stats = match self.input_field_references.get_mut(&type_name) {
+            Some(existing_stats) => existing_stats,
+            None => {
+                self.input_field_references
+                    .insert(type_name.to_string(), HashMap::new());
+                self.input_field_references.get_mut(&type_name).unwrap()
```

Comment:
Could we change the `unwrap()` with an `expect()` please ? Could be done in a follow up PR